src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1507,6 +1503,39 @@ UniqueGenerator<std::string> stringGenerator(const Node& node, GeneratorArgs gen
     return std::make_unique<ConstantAppender<std::string>>(node.to<std::string>());
 }
 
+/**
+ * @param node
+ *   Any node type that evaluates to an array (array literal, literal array of generators, ^Concat,
+ * ^Array)
+ * @return
+ *   The proper generator based on the parsed input.
+ */
+UniqueGenerator<bsoncxx::array::value> bsonArrayGenerator(const Node& node,
+                                                          GeneratorArgs generatorArgs) {
+    // Set of parsers to look when we request an double parser
+    const static std::map<std::string, Parser<UniqueGenerator<bsoncxx::array::value>>> arrayParsers{
+        {"^Concat",
+         [](const Node& node, GeneratorArgs generatorArgs) {
+             return std::make_unique<ConcatGenerator>(node, generatorArgs);
+         }},
+        {"^Array",
```

Comment:
I think you also want LiteralArrayGenerator for ^Verbatim keyword.

Reply:
Added some tests to make sure this works properly.